store/mockstore/unistore/cophandler/closure_exec.go

```diff
@@ -121,6 +157,28 @@ func convertToExprs(sc *stmtctx.StatementContext, fieldTps []*types.FieldType, p
 	return exprs, nil
 }
 
+func getScanExec(dagReq *tipb.DAGRequest) (*tipb.Executor, error) {
+	if len(dagReq.Executors) > 0 {
+		return dagReq.Executors[0], nil
+	}
+	currentExec := dagReq.RootExecutor
+	for currentExec.Tp != tipb.ExecType_TypeTableScan {
```

Comment:
Why not put this check into switch branches?

Reply:
Because `TableScan` executor does not have any child, we need stop the loop once meet `TableScan`